src/app/components/virtualscroller/virtualscroller.ts

```diff
@@ -31,143 +32,127 @@ import { VirtualScrollerLazyLoadEvent } from './virtualscroller.interface';
     template: `
         <div
             [ngClass]="'p-virtualscroller p-component'"
-            [ngStyle]="style"
-            [class]="styleClass"
+            [ngStyle]="style()"
+            [class]="styleClass()"
             [attr.data-pc-name]="'virtualscroller'"
             [attr.data-pc-section]="'root'"
         >
-            <div class="p-virtualscroller-header" *ngIf="header || headerTemplate">
-                <ng-content select="p-header"></ng-content>
-                <ng-container *ngTemplateOutlet="headerTemplate"></ng-container>
-            </div>
+            @if (header() || headerTemplate()) {
+                <div class="p-virtualscroller-header">
+                    <ng-content select="p-header"></ng-content>
+                    <ng-container *ngTemplateOutlet="headerTemplate()"></ng-container>
+                </div>
+            }
             <div #content class="p-virtualscroller-content" [attr.data-pc-section]="'content'">
                 <p-scroller
                     #scroller
-                    [items]="value"
+                    [items]="value()"
                     styleClass="p-virtualscroller-list"
-                    [style]="{ height: scrollHeight }"
-                    [itemSize]="itemSize"
-                    [lazy]="lazy"
+                    [style]="{ height: scrollHeight() }"
+                    [itemSize]="itemSize()"
+                    [lazy]="lazy()"
                     (onLazyLoad)="onLazyItemLoad($event)"
-                    [options]="options"
+                    [options]="options()"
                 >
                     <ng-template #item let-item let-scrollerOptions="options">
-                        <div [ngStyle]="{ height: itemSize + 'px' }" class="p-virtualscroller-item">
+                        <div [ngStyle]="{ height: itemSize() + 'px' }" class="p-virtualscroller-item">
                             <ng-container
                                 *ngTemplateOutlet="
-                                    item ? itemTemplate : loadingItemTemplate;
+                                    item ? itemTemplate() : loadingItemTemplate();
                                     context: { $implicit: item, options: scrollerOptions }
                                 "
                             ></ng-container>
                         </div>
                     </ng-template>
                 </p-scroller>
             </div>
-            <div class="p-virtualscroller-footer" *ngIf="footer || footerTemplate" [attr.data-pc-section]="'footer'">
-                <ng-content select="p-footer"></ng-content>
-                <ng-container *ngTemplateOutlet="footerTemplate"></ng-container>
-            </div>
+            @if (footer() || footerTemplate()) {
+                <div class="p-virtualscroller-footer" [attr.data-pc-section]="'footer'">
```

Comment:
`[attr.data-pc-section]="'footer'"` can be simplified as `data-pc-section="footer"`. It is static